Tests/Realm.Tests/Database/CollectionTests.cs

```diff
@@ -24,2180 +29,2154 @@
-            {
-                Query = query;
-                ExpectedResults = expectedResults;
-            }
+    public readonly struct FtsTestData
+    {
+        public readonly string Query;
+        public readonly string[] ExpectedResults;
 
-            public override readonly string ToString() => Query;
+        public FtsTestData(string query, params string[] expectedResults)
+        {
+            Query = query;
+            ExpectedResults = expectedResults;
         }
 
-        private static readonly object[] FtsTestCases = new object[]
-        {
-            new object[] { new FtsTestData("lord of the", LordOfTheFlies, LordOfTheRings, WheelOfTime, Silmarillion) },
-            new object[] { new FtsTestData("fantasy novel", LordOfTheRings, WheelOfTime) },
-            new object[] { new FtsTestData("popular english", LordOfTheFlies, Silmarillion) },
-            new object[] { new FtsTestData("amazing awesome stuff") },
-            new object[] { new FtsTestData("fantasy -novel", Silmarillion) },
-        };
+        public override string ToString() => Query;
+    }
 
-        [TestCaseSource(nameof(FtsTestCases))]
-        public void Fts_Filter_SimpleTerm(FtsTestData testData)
-        {
-            PopulateFtsData();
+    private static readonly object[] FtsTestCases =
+    {
+        new object[] { new FtsTestData("lord of the", LordOfTheFlies, LordOfTheRings, WheelOfTime, Silmarillion) },
+        new object[] { new FtsTestData("fantasy novel", LordOfTheRings, WheelOfTime) },
+        new object[] { new FtsTestData("popular english", LordOfTheFlies, Silmarillion) },
+        new object[] { new FtsTestData("amazing awesome stuff") },
+        new object[] { new FtsTestData("fantasy -novel", Silmarillion) },
+    };
+
+    [TestCaseSource(nameof(FtsTestCases))]
+    public void Fts_Filter_SimpleTerm(FtsTestData testData)
+    {
+        PopulateFtsData();
 
-            var summaryMatches = _realm.All<ObjectWithFtsIndex>().Filter($"{nameof(ObjectWithFtsIndex.Summary)} TEXT $0", testData.Query).ToArray().Select(o => o.Title);
-            Assert.That(summaryMatches, Is.EquivalentTo(testData.ExpectedResults));
+        var summaryMatches = _realm.All<ObjectWithFtsIndex>().Filter($"{nameof(ObjectWithFtsIndex.Summary)} TEXT $0", testData.Query).ToArray().Select(o => o.Title);
+        Assert.That(summaryMatches, Is.EquivalentTo(testData.ExpectedResults));
 
-            var nullableSummaryMatches = _realm.All<ObjectWithFtsIndex>().Filter($"{nameof(ObjectWithFtsIndex.NullableSummary)} TEXT $0", testData.Query).ToArray().Select(o => o.Title);
-            Assert.That(summaryMatches, Is.EquivalentTo(testData.ExpectedResults));
-        }
+        var nullableSummaryMatches = _realm.All<ObjectWithFtsIndex>().Filter($"{nameof(ObjectWithFtsIndex.NullableSummary)} TEXT $0", testData.Query).ToArray().Select(o => o.Title);
+        Assert.That(nullableSummaryMatches, Is.EquivalentTo(testData.ExpectedResults));
```

Comment:
This was the wrong assertion - even though we were running the query on the nullable field, we weren't validating the results against the expected results.